partner_pricelist_search/tests/test_partner_pricelist_search.py

```diff
@@ -0,0 +1,72 @@
+# Copyright 2021 Tecnativa - Carlos Dauden
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl)
+
+from odoo.exceptions import UserError
+from odoo.tests import common
+
+
+class TestPartnerPricelistSearch(common.SavepointCase):
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.pricelist_1 = cls.env["product.pricelist"].create({
+            "name": "Test pricelist 1",
+        })
+        cls.pricelist_2 = cls.env["product.pricelist"].create({
+            "name": "Test pricelist 2",
+        })
+        cls.customer_1 = cls.env["res.partner"].create({
+            "name": "Test customer 1",
+            "property_product_pricelist": cls.pricelist_1,
+        })
+        cls.customer_2 = cls.env["res.partner"].create({
+            "name": "Test customer 2",
+            "property_product_pricelist": cls.pricelist_2,
+        })
+
+    def test_partner_pricelist_search(self):
+        ResPartner = self.env["res.partner"]
+
+        # Test search '='
+        partners = ResPartner.search([
```

Comment:
You can put each of these checks in a different test method for not having interrupted the execution in case one of them failed, and at the end, there's no initialization needed that forces to share the same method (except the model handler that you can store in a variable in the setup).